File: test/EntityFramework/VBTests/_VbMonsterModel/ExternalTypes/CustomerMm.vb
'---------------------------------------------------------------------
' <copyright file="CustomerMm.vb" company="Microsoft">
'      Copyright (c) Microsoft Corporation.  All rights reserved.
' </copyright>
'
' @owner       avickers
' @backupOwner bricelam
'---------------------------------------------------------------------

Imports System
Imports System.Collections.Generic

Namespace Another.Place
    Partial Public Class CustomerMm
        Public Property CustomerId As Integer
        Public Property Name As String

        Public Property ContactInfo As ContactDetailsMm = New ContactDetailsMm
        Public Property Auditing As AuditInfoMm = New AuditInfoMm

        Public Overridable Property Orders As ICollection(Of OrderMm) = New HashSet(Of OrderMm)
        Public Overridable Property Logins As ICollection(Of LoginMm) = New HashSet(Of LoginMm)
        Public Overridable Property Husband As CustomerMm
        Public Overridable Property Wife As CustomerMm
        Public Overridable Property Info As CustomerInfoMm

    End Class
End Namespace

File: test/EntityFramework/FunctionalTests/_TestModels/TemplateModels/CsMonsterModel/ExternalTypes/CustomerMm.cs
namespace Another.Place
{
    using System.Collections.Generic;
    using FunctionalTests.ProductivityApi.TemplateModels.CsMonsterModel;

    public partial class CustomerMm
    {
        public CustomerMm()
        {
            Orders = new HashSet<OrderMm>();
            Logins = new HashSet<LoginMm>();
            ContactInfo = new ContactDetailsMm();
            Auditing = new AuditInfoMm();
        }

        public int CustomerId { get; set; }
        public string Name { get; set; }

        public ContactDetailsMm ContactInfo { get; set; }
        public AuditInfoMm Auditing { get; set; }

        public virtual ICollection<OrderMm> Orders { get; set; }
        public virtual ICollection<LoginMm> Logins { get; set; }
        public virtual CustomerMm Husband { get; set; }
        public virtual CustomerMm Wife { get; set; }
        public virtual CustomerInfoMm Info { get; set; }
    }
}